crates/zizmor/src/audit/known_vulnerable_actions.rs

```diff
@@ -110,33 +111,77 @@ impl KnownVulnerableActions {
                 _ => Severity::Unknown,
             };
 
-            results.push((severity, vuln.ghsa_id));
+            // Get the first patched version from the first vulnerability in the advisory
+            let first_patched_version = vuln
+                .vulnerabilities
+                .first()
+                .and_then(|v| v.first_patched_version.clone());
+
+            results.push((severity, vuln.ghsa_id, first_patched_version));
         }
 
         Ok(results)
     }
 
+    /// Create a fix to upgrade to a specific non-vulnerable version
+    fn create_upgrade_fix<'doc>(
+        uses: &RepositoryUses,
+        target_version: &str,
+        step: &impl StepCommon<'doc>,
+    ) -> Fix<'doc> {
+        let action_name = format!("{}/{}", uses.owner, uses.repo);
+        let new_uses_value = format!("{action_name}@{target_version}");
```

Comment:
I think we need to specialize this a bit: if `uses:` is pinned by commit hash, then we should resolve `target_version` to a commit and apply that as the patch instead of the version itself. That will also stop us from accidentally introducing a new `unpinned-uses` finding here 🙂 

(And also insert a version comment so that Dependabot and other tools can auto-update it in the future, in which case we might need to use `RewriteFragment` instead of `Replace`.)